pkg/chrootarchive/archive_unix.go

```diff
@@ -5,13 +5,22 @@ package chrootarchive // import "github.com/docker/docker/pkg/chrootarchive"
 
 import (
 	"io"
+	"net"
+	"os/user"
 	"path/filepath"
 	"strings"
 
 	"github.com/docker/docker/pkg/archive"
 	"github.com/pkg/errors"
 )
 
+func init() {
+	// initialize nss libraries in Glibc so that the dynamic libraries are loaded in the host
+	// environment not in the chroot from untrusted files.
+	_, _ = user.Lookup("docker")
+	_, _ = net.LookupHost("localhost")
+}
```

Comment:
FWIW; there were discussions to remove this altogether, as it won't mitigate against the issue. That discussion was not completed though, so moving it (pending the outcome of that) shouldn't hurt.

I marked this for cherry-picking, so that it can get into the 23.0 release